lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'sorbet-runtime'
+require 'concurrent-ruby'
+require 'typhoeus'
+require 'json'
+
+module DevCycle
+  class ConfigManager
+    extend T::Sig
+
+    sig { params(sdkKey: String, local_bucketing: LocalBucketing).returns(NilClass) }
+    def initialize(sdkKey, local_bucketing)
+      @first_load = true
+      @local_bucketing = local_bucketing
+      @sdkKey = sdkKey
+      @config_e_tag = ""
+      @config_poller = Concurrent::TimerTask.new(
+        { execution_interval: @local_bucketing.options.config_polling_interval_ms / 1000, run_now: true }) do |task|
+        fetch_config(false, task)
+      end
+      @config_poller.execute
+      nil
+    end
+
+    def fetch_config(retrying, task)
+      req = Typhoeus::Request.new(
+        get_config_url,
+        headers: {
+          Accept: "application/json",
+        })
+      if @config_e_tag != ""
+        req.headers['If-None-Match'] = @config_e_tag
+      end
+      resp = req.run
+      case resp.code
+      when 304
+        return nil
+      when 200..299
```

Comment:
in Node we only accept 200 responses here: https://github.com/DevCycleHQ/js-sdks/blob/main/sdk/nodejs/src/environmentConfigManager.ts#L104